packages/eas-cli/src/credentials/ios/api/GraphqlClient.ts

```diff
@@ -0,0 +1,233 @@
+        account.id
+      );
+    }
+  }
+
+  public async getDevicesForAppleTeamAsync(
+    { account }: AppLookupParams,
+    { appleTeamIdentifier }: AppleTeam
+  ): Promise<AppleDevice[]> {
+    return await AppleDeviceQuery.getAllByAppleTeamIdentifierAsync(account.id, appleTeamIdentifier);
+  }
+
+  public async createProvisioningProfileAsync(
+    { account }: AppLookupParams,
+    appleAppIdentifier: AppleAppIdentifier,
+    appleProvisioningProfileInput: {
+      appleProvisioningProfile: string;
+      developerPortalIdentifier?: string;
+    }
+  ): Promise<AppleProvisioningProfile> {
+    return await AppleProvisioningProfileMutation.createAppleProvisioningProfileAsync(
+      appleProvisioningProfileInput,
+      account.id,
+      appleAppIdentifier.id
+    );
+  }
+
+  public async getProvisioningProfileAsync(
+    appLookupParams: AppLookupParams,
+    appleTeam: AppleTeam,
+    iosDistributionType: IosDistributionType
+  ): Promise<AppleProvisioningProfile | null> {
+    const projectFullName = formatProjectFullName(appLookupParams);
+    const appleAppIdentifier = await this.createOrGetExistingAppleAppIdentifierAsync(
+      appLookupParams,
+      appleTeam
+    );
+    return await AppleProvisioningProfileQuery.getForAppAsync(projectFullName, {
+      appleAppIdentifierId: appleAppIdentifier?.id,
+      iosDistributionType,
+    });
+  }
+
+  public async updateProvisioningProfileAsync(
+    appleProvisioningProfileId: string,
+    appleProvisioningProfileInput: {
+      appleProvisioningProfile: string;
+      developerPortalIdentifier?: string;
+    }
+  ): Promise<AppleProvisioningProfile> {
+    return await AppleProvisioningProfileMutation.updateAppleProvisioningProfileAsync(
+      appleProvisioningProfileId,
+      appleProvisioningProfileInput
+    );
+  }
+
+  public async getDistributionCertificateForAppAsync(
+    appLookupParams: AppLookupParams,
+    appleTeam: AppleTeam,
+    iosDistributionType: IosDistributionType
+  ): Promise<AppleDistributionCertificate | null> {
+    const projectFullName = formatProjectFullName(appLookupParams);
+    const appleAppIdentifier = await this.createOrGetExistingAppleAppIdentifierAsync(
+      appLookupParams,
+      appleTeam
+    );
+    return await AppleDistributionCertificateQuery.getForAppAsync(projectFullName, {
+      appleAppIdentifierId: appleAppIdentifier?.id,
+      iosDistributionType,
+    });
+  }
+
+  public async getDistributionCertificatesForAccountAsync({
+    account,
+  }: AppLookupParams): Promise<AppleDistributionCertificate[]> {
+    return await AppleDistributionCertificateQuery.getAllForAccount(account.name);
+  }
+
+  public async createDistributionCertificateAsync(
+    { account }: AppLookupParams,
+    distCert: DistributionCertificate
+  ): Promise<AppleDistributionCertificate> {
+    const appleTeam = nullthrows(
+      await AppleTeamQuery.byAppleTeamIdentifierAsync(account.id, distCert.teamId)
```

Comment:
we can use the `createOrGetExisting` AppleTeam helper you wrote here. It is possible we haven't registered the AppleTeam on the expo servers yet.